src/java_bytecode/library/src/java/lang/Thread.java

```diff
@@ -0,0 +1,1980 @@
+      millis++;
+    }
+
+    sleep(millis);
+  }
+
+  /**
+   * Initializes a Thread with the current AccessControlContext.
+   * @see #init(ThreadGroup,Runnable,String,long,AccessControlContext,boolean)
+   */
+  private void init(ThreadGroup g, Runnable target, String name, long stackSize)
+  {
+    CProver.notModelled();
+    //init(g, target, name, stackSize, null, true);
+  }
+
+  // /**
+   // * Initializes a Thread.
+   // *
+   // * @param g the Thread group
+   // * @param target the object whose run() method gets called
+   // * @param name the name of the new Thread
+   // * @param stackSize the desired stack size for the new thread, or
+   // *        zero to indicate that this parameter is to be ignored.
+   // * @param acc the AccessControlContext to inherit, or
+   // *            AccessController.getContext() if null
+   // * @param inheritThreadLocals if {@code true}, inherit initial values for
+   // *            inheritable thread-locals from the constructing thread
+   // */
+  // private void init(ThreadGroup g, Runnable target, String name,
+                    // long stackSize, AccessControlContext acc,
+                    // boolean inheritThreadLocals) {
+    // if (name == null)
+    // {
+      // throw new NullPointerException("name cannot be null");
+    // }
+    // this.name = name;
+    // this.target = target;
+    // tid = ++threadSeqNumber;
+
+    // // Indicate this function is not fully modelled
+    // CProver.notModelled();
+  // }
+
+  /**
+   * Throws CloneNotSupportedException as a Thread can not be meaningfully
+   * cloned. Construct a new Thread instead.
+   *
+   * @throws  CloneNotSupportedException
+   *          always
+   */
+  //@Override
+  protected Object clone() throws CloneNotSupportedException {
+    throw new CloneNotSupportedException();
+  }
+
+  /**
+   * Allocates a new {@code Thread} object. This constructor has the same
+   * effect as {@linkplain #Thread(ThreadGroup,Runnable,String) Thread}
+   * {@code (null, null, gname)}, where {@code gname} is a newly generated
+   * name. Automatically generated names are of the form
+   * {@code "Thread-"+}<i>n</i>, where <i>n</i> is an integer.
+   */
+  public Thread()
+  {
+    target = this;
+    tid = ++threadSeqNumber;
+  }
+
+  /**
+   * Allocates a new {@code Thread} object. This constructor has the same
+   * effect as {@linkplain #Thread(ThreadGroup,Runnable,String) Thread}
+   * {@code (null, target, gname)}, where {@code gname} is a newly generated
+   * name. Automatically generated names are of the form
+   * {@code "Thread-"+}<i>n</i>, where <i>n</i> is an integer.
+   *
+   * @param  target
+   *         the object whose {@code run} method is invoked when this thread
+   *         is started. If {@code null}, this classes {@code run} method does
+   *         nothing.
+   */
+  public Thread(Runnable target)
+  {
+    this.target = target;
+    tid = ++threadSeqNumber;
```

Comment:
Datarace on the increment of this variable. Use the original function to increment.

Reply:
done